handler.go

```diff
@@ -41,6 +42,9 @@ func (h *httpHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	h.handler.Serve(ctx, r.Body, &resp)
 
+	io.Copy(ioutil.Discard, r.Body) // Ignoring error since r.Body may already be closed
```

Comment:
This potentially discards customer data without giving the fdk user any warning that there was excess data and that it was thrown away.  Minimally, it seems like a warning would required although if it were up to me, I would make this an assertion type error.